### PR TITLE
add com.github.swilmet.gcsvedit.yml

### PR DESCRIPTION
**Pull request by sonnyp**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/swilmet/gCSVedit/issues/6
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by bilelmoussaoui** [comment]
bot, build com.github.swilmet.gcsvedit

File: com.github.swilmet.gcsvedit.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+  # Needed for GVfs metadata to work
+  - "--talk-name=org.gtk.vfs.*"
+
+build-options:
+  cflags: "-O2 -g"
+  cxxflags: "-O2 -g"
+  env:
+    V: '1'
+
```
Comment (barthalion) [inline com.github.swilmet.gcsvedit.yml:27]:
```suggestion
```

File: com.github.swilmet.gcsvedit.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+finish-args:
+  # X11 + XShm access
+  - "--share=ipc"
+  - "--socket=fallback-x11"
+
+  # Wayland access
+  - "--socket=wayland"
+
+  # We want full fs access so we can read the files
+  - "--filesystem=host"
```
Comment (barthalion) [inline com.github.swilmet.gcsvedit.yml:17]: Ideally it should use GtkFileChooserNative to use portals.

**Comment by nedrichards** [comment]
bot, build com.github.swilmet.gcsvedit

**Comment by sonnyp** [comment]
The build is failing because gCSVEdit doesn't ship an icon, it uses `accessories-text-editor` https://github.com/swilmet/gCSVedit/blob/0.8.0/data/com.github.swilmet.gcsvedit.desktop#L7

I will send a PR upstream.

Is there a way I can run the flathub test suite myself (or at least a subset), ideally on my computer ?

**Comment by barthalion** [comment]
You can run appstream validation locally with `org.freedesktop.appstream-glib`, the icon availability check simply looks for 128x128 icon matching the app-id in `/app/share/app-info`.

File: com.github.swilmet.gcsvedit.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
+---
+app-id: com.github.swilmet.gcsvedit
+runtime: org.gnome.Platform
+runtime-version: '3.38'
```
Comment (hfiguiere) [inline com.github.swilmet.gcsvedit.yml:4]: we can now use GNOME 40. It would be better.

**Review by hfiguiere — COMMENTED** [review]
You could add the patch from the Pull Request for the icon.

**Comment by barthalion** [comment]
Looks like gcsvedit was removed from GitHub so I'll just close it.

**Pull request closed without merging**